src/data_structures/fmindex.rs

```diff
@@ -328,55 +376,78 @@ impl FMDIndex {
         self.backward_ext(&interval.swapped(), comp_a)
             .swapped()
     }
-
-    /// Provide a reference to the underlying FMIndex.
-    pub fn fmindex(&self) -> &FMIndex {
-        &self.fmindex
-    }
 }
 
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use alphabets::dna;
-    use data_structures::suffix_array::suffix_array;
-    use data_structures::bwt::bwt;
+    use data_structures::suffix_array::{suffix_array, SuffixArray};
+    use data_structures::bwt::{bwt, less, Occ};
+    use std::rc::Rc;
+
+    fn sa_interval_to_vec(sa: &SuffixArray, interval: &Interval) -> Vec<usize> {
```

Comment:
@johanneskoester Looking at the doctests, it seems like this is the primary use of intervals (at least for the trivial examples). Should I add this as a method on `Interval` taking a suffix array as an argument (probably also using a proper trait bound rather than my trait object reference)?